.github/workflows/linux.yml

```
@@ -25,8 +25,26 @@ jobs:
       run: >
        sudo apt-get update && sudo apt install libx11-dev libxext-dev libwayland-dev libfuse2 clang build-essential
 
+    - name: Cache CMake dependency source code
+      uses: actions/cache@v4
+      env:
+          cache-name: cache-cmake-dependency-sources
+      with:
+          path: | 
+            ${{github.workspace}}/build
+          key: ${{ env.cache-name }}-sdl-${{ hashFiles('**/CMakeLists.txt', 'cmake/**') }}
+          restore-keys: |
+            ${{ env.cache-name }}-sdl-
+
+    - name: Cache CMake dependency build objects
+      uses: hendrikmuhs/ccache-action@v1.2.14
+      env:
+          cache-name: cache-cmake-dependency-builds
+      with:
+        key: ${{ env.cache-name }}-${{ matrix.os }}-${{ matrix.cxx }}-sdl
```

I've discovered a small mistake on the cache naming, it was trying to find `matrix.os` and `matrix.cxx` which are 2 variables set at the beginning as `jobs: build: matrix: os/cxx`, used only when multiple OSes are running in the same workflow file, but here they are unused.
To fix that just need to put `runner.os` over the rest. This need to be done on all files.
```suggestion
        key: ${{ env.cache-name }}-${{ runner.os }}-sdl
```